2022/incoming/lapland-packaging-department-pt1.pod

```diff
@@ -0,0 +1,113 @@
+
+There was not much time left until Advent so he decided to stick with Perl, his favourite programming language.
+
+- Perl is capable of everything by itself, anyway - he thought.
+
+Unfortunately there was no chance to use tools like cpan or cpanm only on that single machine connected to the internet.
+And that connection worked only when the C<http_proxy> and C<https_proxy> were explicitly set beforehand. 
+
+First of all he needed a solution to be able to use at least a handful of modules to accomplish his work.
+
+- Why not compile the required modules from source downloaded from CPAN? - he asked himself.
+
+It sounded like a good starting point. He immediately made his hands dirty. 
+
+Tonttu quickly jotted down a small how-to: 
+
+This method can be used to compile any Perl module manually on Linux!
+
+First of all make sure, that C<PERL5LIB> is loaded, if not load it with export C<PERL5LIB=/path/to/your/perl> (especially if you have Perl installed to a custom location).
+
+=begin code 
+
+$ echo $PERL5LIB
+/home/cabox/perl5/lib/perl5
+
+=end code 
+
+If the output is empty, that means PERL5lib is not loaded (most probably Perl is installed into a custom location). So let's load it:
+
+=begin code 
+
+export PERL5LIB=/path/to/your/perl
+
+=end code 
+
+Second  we need to download the module from CPAN.
+Let's demonstrate the process with Pod::Advent:
+
+Download the tarball for the installation:
+
+=begin code 
+
+$ wget https://cpan.metacpan.org/authors/id/D/DA/DAVIDRW/Pod-Advent-0.24.tar.gz
+
+=end code 
+
+Untar the archive:
+
+=begin code 
+
+tar -xzvf Pod-Advent-0.24.tar.gz
+
+=end code 
+
+Switch to the directory of the module's source:
+
+=begin code 
+
+$ cd Pod-Advent-0.24
+
+=end code 
+
+Execute Makefile.PL with Perl.
+
+=begin code 
+
+$ perl Makefile.PL
+Checking if your kit is complete...
+Looks good
+Generating a Unix-style Makefile
+Writing Makefile for Pod::Advent
+Writing MYMETA.yml and MYMETA.json
+
+=end code 
+
+Then execute C<make> and C<make install>.
+
+=begin code 
+
+$ make
+cp lib/Pod/Advent.pm blib/lib/Pod/Advent.pm
+cp bin/pod2advent blib/script/pod2advent
+"/home/cabox/perl5/perlbrew/perls/perl-5.28.0/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/pod2advent
+Manifying 1 pod document
+Manifying 1 pod document
+
+$ make install                                                                       
+Manifying 1 pod document
+Manifying 1 pod document
+Appending installation info to /home/cabox/perl5/lib/perl5/x86_64-linux/perllocal.pod
+
+=end code 
+
+In case of a more complicated module tests might fail, either You might be presented with errors in the output, or worst case You might face the dependency hell.
+DISCLAIMER: To avoid dependency hell or other errors upon module installation it is recommended to use cpan or cpanm wherever and whenever possible.
```

Comment:
```suggestion
In the case of a more complicated module, tests might fail. You might be presented with errors in the output, or in the worst case, you might face dependency hell.
DISCLAIMER: To avoid dependency hell or other errors upon module installation it is recommended to use L<cpan> or L<cpanm> wherever and whenever possible.
```